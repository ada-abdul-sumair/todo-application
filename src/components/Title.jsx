// React
import React from 'react';

// Styles
import { Stack, Typography, Divider } from '@mui/material';

// Icons
import ChecklistIcon from '@mui/icons-material/Checklist';

export default function Title() {
  return (
    <Stack
      direction="row"
      spacing={2}
      alignItems="center"
      divider={<Divider orientation="vertical" />}
    >
      <ChecklistIcon style={{ fontSize: 100, color: '#353b48' }} />
      <Typography
        data-testid="main-page-title"
        style={{ fontSize: 50, fontWeight: 'bold', color: '#353b48' }}
      >
        TODO LIST
      </Typography>
    </Stack>
  );
}
