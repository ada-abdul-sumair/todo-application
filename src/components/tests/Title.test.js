import React from 'react';
import { render, screen } from '@testing-library/react';
import Title from '../Title';

let asFragment;

describe('Title Component', () => {
  beforeEach(() => {
    ({ asFragment } = render(<Title />));
  });
  // Rendering Test
  test('renders key elements', () => {
    const checklistIcon = screen.getByTestId('checklist-icon');
    expect(checklistIcon).toBeInTheDocument();

    const mainTitle = screen.getByTestId('main-page-title');
    expect(mainTitle).toBeInTheDocument();
    expect(screen.getByTestId('main-page-title')).toHaveTextContent(
      'TODO LIST'
    );
  });

  // Snapshot Test
  test('matches snapshot', () => {
    expect(asFragment()).toMatchSnapshot(); //delete __snapshots__ folder to update it if tests fail
  });

  // Style Test
  test('has the correct styles', () => {
    expect(screen.getByTestId('main-page-title')).toHaveStyle(
      'font-size: 50px'
    );
  });
});
