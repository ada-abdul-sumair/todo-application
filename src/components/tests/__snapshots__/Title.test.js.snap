// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1sos3zc-MuiStack-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="checklist-icon"
      focusable="false"
      style="font-size: 100px;"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 7h-9v2h9V7zm0 8h-9v2h9v-2zM5.54 11 2 7.46l1.41-1.41 2.12 2.12 4.24-4.24 1.41 1.41L5.54 11zm0 8L2 15.46l1.41-1.41 2.12 2.12 4.24-4.24 1.41 1.41L5.54 19z"
      />
    </svg>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-ftc7wc-MuiDivider-root"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      data-testid="main-page-title"
      style="font-size: 50px;"
    >
      Todo List
    </p>
  </div>
</DocumentFragment>
`;
